pkg/controller/intrusiondetection/intrusiondetection_controller.go

```diff
@@ -204,6 +220,13 @@ func (r *ReconcileIntrusionDetection) Reconcile(ctx context.Context, request rec
 	r.status.OnCRFound()
 	reqLogger.V(2).Info("Loaded config", "config", instance)
 
+	// Set defaults for IntrusionDetection resource
+	if err := r.setDefaultsOnIntrusionDetection(ctx, instance); err != nil {
+		reqLogger.V(3).Info("failed to set defaults on IntrusionDetection CR", "err", err)
```

Comment:
I think this should just be an error, if we failed to set the defaults then a real error occurred and should be logged (as opposed to we're waiting for something to become available and we degrade).